### PR TITLE
feat(chat): Make chat push content to the side in tile view

### PR DESCRIPTION
**Pull request by muscat1**
First part of a two-part feature that makes chat push the content to the side rather than display over it. The large-view implementation is underway.

### CODE REVIEW

**Comment by muscat1** [comment]
Chat

File: css/filmstrip/_tile_view.scss [hunk 1]
```diff
@@ -45,7 +46,18 @@
...
+        z-index: $filmstripVideosZ;
+        margin-left: 0px;
+        transition: margin 0.5s, width 0.5s;
+
+        &.shift-right {
+            margin-left: $sidebarWidth;
+            width: calc(100% - #{$sidebarWidth});
+
+            #filmstripRemoteVideos {
+                width: calc(100vw - #{$sidebarWidth});
```
Comment (hristoterezov) [inline css/filmstrip/_tile_view.scss]: why `100vw` and not `100%`?

File: react/features/chat/components/AbstractChat.js [hunk 2]
```diff
@@ -86,6 +92,17 @@ export function _mapDispatchToProps(dispatch: Dispatch<any>) {
...
+
+        /**
+         * Recomputes the client size taking chat size into account, such that the content is shifted.
+         *
+         * @param {number} clientWidth - The new width of the window.
+         * @param {number} clientHeight - The new height of the window.
+         * @returns {Function}
+         */
+        _resizeWindow(clientWidth: number, clientHeight: number) {
+            dispatch(clientResized(clientWidth, clientHeight));
```
Comment (hristoterezov) [inline react/features/chat/components/AbstractChat.js]: hmm this action is not supposed to be used for that. It is supposed to really store the size of the window.

File: css/filmstrip/_tile_view.scss (hunk 1 above)
Comment (muscat1) [reply css/filmstrip/_tile_view.scss]: 100vw is the width initially set on `#filmstripRemoteVideos` and I thought I'd preserve that, but I can change it if you consider it's more appropriate.

File: react/features/chat/components/AbstractChat.js (hunk 2 above)
Comment (muscat1) [reply react/features/chat/components/AbstractChat.js]: Well from what I've looked into, and also discussed with @saghul this is the main way of recomputing the size of the window in order to properly display the tiles and in this situation, the area in which we want to display the tiles actually has to shrink by the chat width, so it seemed natural to use this method, but I'm open to any other suggestion.

File: css/filmstrip/_tile_view.scss (hunk 1 above)
Comment (hristoterezov) [reply css/filmstrip/_tile_view.scss]: No that's fine! Make sense!

File: react/features/chat/components/AbstractChat.js (hunk 2 above)
Comment (hristoterezov) [reply react/features/chat/components/AbstractChat.js]: I see this is probably the easiest way but we want to use this for a window resizes and to store the window size not the available size for the tile view area.

Proposal:
`setTileViewDimensions` should take into account somehow if the chat is opened or not. Maybe by parameter or you can convert it to async action in order to access the state. You can pass to it the size of the available space for the tiles after calculating it. Please check all the places where we call this method. Also check react/features/filmstrip/middleware.web.js and react/features/filmstrip/subscriber.web.js. You will need to `setTileViewDimensions` when the chat opens or closes like we do for other cases (resize of the window, entering participants, etc).

**Comment by saghul** [comment]
What's the state of this?

**Comment by muscat1** [comment]
There seem to be small UI issues when putting together this PR and the large view one https://github.com/jitsi/jitsi-meet/pull/5148 I've been caught up in some other stuff but I'll take a look and try to sort them out sometime tomorrow or early next week :D

**Comment by luixxiul** [comment]
This needs rebase.

**Pull request merged**